waspc/cli/src/Wasp/Cli/Interactive.hs

```diff
@@ -0,0 +1,86 @@
+module Wasp.Cli.Interactive
+  ( askForInput,
+    askToChoose,
+    askForRequiredInput,
+  )
+where
+
+import Data.Foldable (find)
+import Data.List (intercalate)
+import qualified Data.Text as T
+import System.IO (hFlush, stdout)
+import Text.Read (readMaybe)
+import qualified Wasp.Util.Terminal as Term
+
+askForRequiredInput :: String -> IO String
+askForRequiredInput question = do
+  ensureNotNull . askForInput $ question
+
+askToChoose :: Show a => String -> [a] -> IO a
```

Comment:
NOTE: I left some comments below, but ended up writing a whole new implementation fro this function for you to check out, as we discussed, so check the comments on this functions for idea what I thought can be improved, and then at the end of the function see the new implementation that should improve that stuff + some more.